core/rend/vulkan/pipeline.h

```diff
@@ -86,38 +86,36 @@ class DescriptorSets
 			perPolyDescSet = perPolyAlloc.alloc();
 			std::vector<vk::WriteDescriptorSet> writeDescriptorSets;
 
-			vk::DescriptorImageInfo imageInfo;
 			if (poly.texture != nullptr)
 			{
-				imageInfo = vk::DescriptorImageInfo(samplerManager->GetSampler(poly.tsp),
+				vk::DescriptorImageInfo imageInfo(samplerManager->GetSampler(poly.tsp),
 						((Texture *)poly.texture)->GetReadOnlyImageView(), vk::ImageLayout::eShaderReadOnlyOptimal);
-				writeDescriptorSets.emplace_back(perPolyDescSet, 0, 0, 1, vk::DescriptorType::eCombinedImageSampler, &imageInfo, nullptr, nullptr);
+				writeDescriptorSets.emplace_back(perPolyDescSet, 0, 0, vk::DescriptorType::eCombinedImageSampler, imageInfo);
 			}
 
-			vk::DescriptorBufferInfo uniBufferInfo;
-			vk::DescriptorBufferInfo lightBufferInfo;
 			if (poly.isNaomi2())
 			{
 				const vk::DeviceSize uniformAlignment = VulkanContext::Instance()->GetUniformBufferAlignment();
 				size_t size = sizeof(N2VertexShaderUniforms) + align(sizeof(N2VertexShaderUniforms), uniformAlignment);
-				uniBufferInfo = vk::DescriptorBufferInfo{ buffer, uniformOffset + polyNumber * size, sizeof(N2VertexShaderUniforms) };
-				writeDescriptorSets.emplace_back(perPolyDescSet, 2, 0, 1, vk::DescriptorType::eUniformBuffer, nullptr, &uniBufferInfo, nullptr);
+				vk::DescriptorBufferInfo uniBufferInfo(buffer, uniformOffset + polyNumber * size, sizeof(N2VertexShaderUniforms));
+				writeDescriptorSets.emplace_back(perPolyDescSet, 2, 0, vk::DescriptorType::eUniformBuffer, nullptr, uniBufferInfo);
 
 				size = sizeof(N2LightModel) + align(sizeof(N2LightModel), uniformAlignment);
+				vk::DescriptorBufferInfo lightBufferInfo;
 				// light at index 0 is no light
 				if (poly.lightModel != nullptr)
 					lightBufferInfo = vk::DescriptorBufferInfo{ buffer, lightOffset + (poly.lightModel - pvrrc.lightModels.head() + 1) * size, sizeof(N2LightModel) };
 				else
 					lightBufferInfo = vk::DescriptorBufferInfo{ buffer, lightOffset, sizeof(N2LightModel) };
-				writeDescriptorSets.emplace_back(perPolyDescSet, 3, 0, 1, vk::DescriptorType::eUniformBuffer, nullptr, &lightBufferInfo, nullptr);
+				writeDescriptorSets.emplace_back(perPolyDescSet, 3, 0, vk::DescriptorType::eUniformBuffer, nullptr, lightBufferInfo);
```

Comment:
uniBufferInfo and lightBufferInfo are used by reference and go out of scope

Reply:
Fixed